app/src/main/java/org/jellyfin/androidtv/ui/playback/WatchTrackerViewModel.kt

```diff
@@ -0,0 +1,109 @@
+package org.jellyfin.androidtv.ui.playback
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.jellyfin.androidtv.preference.UserPreferences
+import org.jellyfin.androidtv.ui.playback.stillwatching.StillWatchingFragment
+import org.jellyfin.androidtv.ui.playback.stillwatching.StillWatchingStates
+import org.jellyfin.androidtv.util.TimeUtils.MILLIS_PER_MIN
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import timber.log.Timber
+
+
+class WatchTrackerViewModel : ViewModel(), KoinComponent {
+	private lateinit var videoManager: VideoManager
+	private var episodeCount = 0
+	private var watchTime = 0L
+	private var lastUpdateTime = 0L
+	private var itemWasInterrupted: Boolean = false
+	private var itemIsEpisode: Boolean = false
+
+	private val userPreferences by inject<UserPreferences>()
+
+	fun getItemIsEpisode(): Boolean {
+		return itemIsEpisode
+	}
+
+	fun getItemWasInterrupted(): Boolean {
+		return itemWasInterrupted
+	}
+
+	fun updateLastUpdateTime() {
+		lastUpdateTime = System.currentTimeMillis()
+	}
+
+	fun onEpisodeWatched() {
+		Timber.i("Watcher onEpisodeWatched")
+		if (!itemWasInterrupted) episodeCount++
+		itemWasInterrupted = false
+	}
+
+
+	fun onUserInteraction() {
+		Timber.i("Watcher onUserInteraction")
+		resetWatchTime()
+		itemWasInterrupted = true
+	}
+
+
+	fun startWatchTime() {
+		Timber.i("Start tracker")
+		resetWatchTime()
+		updateLastUpdateTime()
+		itemIsEpisode = true
+		itemWasInterrupted = false
+	}
+
+	fun setVideoManager(videoManager: VideoManager) {
+		this.videoManager = videoManager
+	}
+
+	fun updateWatchTime() {
+		calculateWatchTime()
+		checkPrompt()
+	}
+
+
+	private fun checkPrompt() {
+		if (!::videoManager.isInitialized) return
+
+		val stillWatchingSetting = StillWatchingStates.getSetting(userPreferences[UserPreferences.stillWatchingEnabled].toString())
+
+		val currentEpisodeProgress = videoManager.currentPosition.toFloat() / videoManager.duration.toFloat()
+		val minMinutesInMs = stillWatchingSetting.minMinutes * MILLIS_PER_MIN
+
+		// At episode count, your watch time is above min minute threshold, and you are at least 10% of the way through the next episode
+		val episodeRequirementMet = episodeCount == stillWatchingSetting.episodeCount && watchTime >= minMinutesInMs && currentEpisodeProgress >= 0.1
+
+		// Above min minute threshold and you are not on the episode to meet episode requirement
```

Comment:
Can you explain `and you are not on the episode to meet episode requirement`? This feels superfluous